test/language/statements/for-of/iterator-close-throw-get-method-non-callable.js

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2020 Alexey Shvayka. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-iteratorclose
+description: >
+  If retrieving an iterator's `return` method generates an error while
+  closing the iterator with throw completion, this error should be suppressed.
+info: |
+  IteratorClose ( iteratorRecord, completion )
+
+  [...]
+  4. Let innerResult be GetMethod(iterator, "return").
+  5. If innerResult.[[Type]] is normal,
+    [...]
+  6. If completion.[[Type]] is throw, return Completion(completion).
+  7. If innerResult.[[Type]] is throw, return Completion(innerResult).
+
+  GetMethod ( V, P )
+
+  [...]
+  2. Let func be ? GetV(V, P).
+  3. If func is either undefined or null, return undefined.
+  4. If IsCallable(func) is false, throw a TypeError exception.
+features: [Symbol.iterator]
+---*/
+
+var iterable = {};
+var iterationCount = 0;
+
+iterable[Symbol.iterator] = function() {
+  return {
+    next: function() {
+      return { done: false, value: null };
+    },
+    return: 'str',
```

Comment:
This test is incorrect. GetMethod() will throw a TypeError because `'str'` is not callable, which shadows the error in the loop.

Reply:
The proposed spec is:

```
4. Let innerResult be GetMethod(iterator, "return").
5. If innerResult.[[Type]] is normal,
     [...]
6. If completion.[[Type]] is throw, return Completion(completion).
7. If innerResult.[[Type]] is throw, return Completion(innerResult).
```

`innerResult.[[Type]]` would be `throw`, so we are skipping step 5.
`completion.[[Type]]` is also `throw`, so we are returning original error from the loop in step 6.

Reply:
It doesn't feel right to me to override user-provided error, and behaviour overview of #1408 does not mention this, I was just following the normative text.

Reply:
ah i didn't realize these tests were for something which isn't part of the spec yet... they should probably be tagged with a feature or something.

Reply:
test262 tests generally match the spec + stage 3 + has-consensus PRs.